### PR TITLE
Use Chrome manifest V3

### PR DESCRIPTION
**Pull request by yushih**
https://emurgo.atlassian.net/browse/YOEXT-125

See the comment of https://emurgo.atlassian.net/wiki/spaces/YE/pages/739311617/Checklist+to+upgrade+to+manifest+v3 for documentation.

## Notes to developers:

Originally, the dev server is started by running:
```
npm run dev:stable
```
Now, this command still works for backward compatibility, but it is recommend to run
```
npm run dev:background
```
and
```
npm run dev:main
```
separately in two consoles. 

## Notes to release manager:
The originally commands all build mv3 packages.
To build mv2 packages (for Firefox), use:
```
npm run prod:nightly-mv2
```
and
```
prod:stable-mv2
```

### CODE REVIEW

**Comment by neuodev** [comment]
The extension size is unreasonably big. it is **107 MB**!! The current extension size is **< 1 MB** (dev). The final nightly size is about **30 MB**.
in the `dev/js` directory, there are a lot of assets/wasm files. Is there any reason for this?!
<details>
<summary> Details </summary>
Screen Shot 2022-08-30 at 3 03 20 PM
<br /> <br /> <br />
Screen Shot 2022-08-30 at 3 02 51 PM
</details>

**Comment by neuodev** [comment]
@yushih  For some reason, my terminal doesn't work when I try to exit the server while using `stumx`. I am using `bash` on MaxOS.  I don't know if I have a problem with my terminal but we can use [Concurrently](https://www.npmjs.com/package/concurrently) for the same use case.

**Comment by zuzunker** [comment]
> The extension size is unreasonably big. it is **107 MB**!! The current extension size is **< 1 MB** (dev). The final nightly size is about **30 MB**.
> in the `dev/js` directory, there are a lot of assets/wasm files. Is there any reason for this?!

@AhmedIbrahim336 , when I try building the extension zip file, I'm getting 26.6 megabytes, which is roughly the same as in production at the moment.

**Comment by neuodev** [comment]
> @AhmedIbrahim336 , when I try building the extension zip file, I'm getting 26.6 megabytes, which is roughly the same as in production at the moment.

@vsubhuman  What about when you unzip it? The current size of the unzipped nightly build is about **50MB**

**Comment by yushih** [comment]
@AhmedIbrahim336 the bloating of `/dev` dir size is expected. Because originally, the dir only had to hold `background.html`, and all chunks of the background script were served by webpack-dev-server from memory. But now that mv3 allows only a JS background service worker module, it has to be written to disk (`writeToDisk: true` in `webpack/devConfig.js`).

**Comment by yushih** [comment]
I'll take a look into the production bundle size.

**Comment by neuodev** [comment]
> @AhmedIbrahim336 the bloating of /dev dir size is expected. Because originally, the dir only had to hold background.html, and all chunks of the background script were served by webpack-dev-server from memory. But now that mv3 allows only a JS background service worker module, it has to be written to disk (writeToDisk: true in webpack/devConfig.js).

@yushih  Got it. If we can bundle all these assets into a separate directory this will be much cleaner.

**Review by zuzunker — COMMENTED** [review]
/check

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Review by zuzunker — COMMENTED** [review]
/check

**Review by a-hosani — APPROVED** [review]

**Review by zuzunker — COMMENTED** [review]
/check

**Review by zuzunker — COMMENTED** [review]
/check

**Review by zuzunker — COMMENTED** [review]
/check

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Review by a-hosani — APPROVED** [review]

**Comment by vsubhuman** [comment]
https://emurgo.atlassian.net/browse/YOEXT-125

**Comment by neuodev** [comment]
[autoscript] ⚠️ `yushi/manifest-v3` doesn't full our branch naming convention.
Should follow `<tag>/<jira-ticket-id>/<short-name>`
<details>
<summary>Details</summary>

- `<tag>`: `fix`, `feat`, `chore`, `refactor`, `test`, ...
- `<jira-ticket-id>`: The issue ID of JIRA all capital letters
- `<short-name>`: A short and descriptive name for the branch
</details>

[](__JIRA_COMMENT_FLAG)

File: packages/yoroi-connector/src/inject.js [hunk 1]
```diff
@@ -33,10 +33,10 @@ async function injectIntoPage(code) {
         container.removeChild(scriptTag);
         console.log(`[yoroi/${YOROI_TYPE}] dapp-connector is successfully injected into ${location.hostname}`);
         markInjectionInDocument(container);
-        return true;
+        resolve(true);
     } catch (e) {
         console.error(`[yoroi/${YOROI_TYPE}] injection failed!`, e);
-        reject(e);
+        resolve(false);
```
Comment (yushih) [inline packages/yoroi-connector/src/inject.js:42]: Yes. This change is valid. `reject(e)` was wrong.

File: packages/yoroi-connector/src/inject.js [hunk 2]
```diff
@@ -33,10 +33,10 @@ async function injectIntoPage(code) {
         container.removeChild(scriptTag);
         console.log(`[yoroi/${YOROI_TYPE}] dapp-connector is successfully injected into ${location.hostname}`);
         markInjectionInDocument(container);
-        return true;
+        resolve(true);
```
Comment (yushih) [inline packages/yoroi-connector/src/inject.js]: This change is problematic. I forgot to remove the `return true` statement. But the promise should not be resolved here. See `resolveScriptedInject`.

Comment (zuzunker) [reply packages/yoroi-connector/src/inject.js]: ahhh, understood, I will update 👌

**Review by zuzunker — APPROVED** [review]

**Review by zuzunker — COMMENTED** [review]
/check

**Pull request merged**